cmd/kafka-consumer/writer.go

```diff
@@ -542,22 +545,31 @@ func (w *writer) appendRow2Group(dml *commonEvent.DMLEvent, progress *partitionP
 		// we cannot distinguish whether it's a real fallback event or not, still append it.
 		if w.partitionTableAccessor.IsPartitionTable(schema, table) {
 			log.Warn("DML events fallback, but it's canal-json or open-protocol and the table is a partition table, still append it",
-				zap.Int32("partition", group.partition), zap.Any("offset", offset),
-				zap.Uint64("commitTs", commitTs), zap.Uint64("highWatermark", group.highWatermark),
+				zap.Int32("partition", group.Partition), zap.Any("offset", offset),
+				zap.Uint64("commitTs", commitTs), zap.Uint64("highWatermark", group.HighWatermark),
 				zap.String("schema", schema), zap.String("table", table), zap.Int64("tableID", tableID),
 				zap.Stringer("eventType", dml.RowTypes[0]))
 			group.Append(dml, true)
 			return
 		}
 		log.Warn("DML event fallback row, since less than the group high watermark, ignore it",
 			zap.Int32("partition", progress.partition), zap.Any("offset", offset),
-			zap.Uint64("commitTs", commitTs), zap.Uint64("highWatermark", group.highWatermark),
+			zap.Uint64("commitTs", commitTs), zap.Uint64("HighWatermark", group.HighWatermark),
 			zap.Any("partitionWatermark", progress.watermark), zap.Any("watermarkOffset", progress.watermarkOffset),
 			zap.String("schema", schema), zap.String("table", table), zap.Int64("tableID", tableID),
 			zap.Stringer("eventType", dml.RowTypes[0]),
 			// zap.Any("columns", row.Columns), zap.Any("preColumns", row.PreColumns),
 			zap.Any("protocol", w.protocol), zap.Bool("IsPartition", dml.TableInfo.TableName.IsPartition))
 	default:
+		if w.enableTableAcrossNodes {
```

Comment:
I'm confused when will the code into default branch? So how can the code handle when the protocol is ProtocolCanalJSON and also w.enableTableAcrossNodes is true